docs/design/ovs-pipeline.md

```diff
@@ -913,7 +1099,32 @@ but is also harmless, as the destination MAC is already correct.
 
 Flow 2 commits all other new connections.
 
-All traffic then goes to the next table ([L2ForwardingOutTable]).
+All traffic then goes to the next table ([HairpinSNATTable]).
+
+### HairpinSNATTable (108)
+
+The table is used to handle hairpin Service packet, which indicates that the
+packet should be output on the port on which it was received.
+
+If you dump the flows for this table, you should see the flows:
+
+```text
+1. table=108, priority=200,ip,nw_src=10.10.0.4,nw_dst=10.10.0.4 actions=mod_nw_src:169.254.169.252,load:0x1->NXM_NX_REG0[18],resubmit(,110)
+2. table=108, priority=200,ip,nw_src=10.10.0.2,nw_dst=10.10.0.2 actions=mod_nw_src:169.254.169.252,load:0x1->NXM_NX_REG0[18],resubmit(,110)
+3. table=108, priority=200,ip,nw_src=10.10.0.3,nw_dst=10.10.0.3 actions=mod_nw_src:169.254.169.252,load:0x1->NXM_NX_REG0[18],resubmit(,110)
+4. table=108, priority=0 actions=resubmit(,110)
+```
+
+Flow 1-3 are used to match Service packet from Pods. The source IP of matched
+packet should be SNAT'd with a virtual hairpin IP since the source and destination
+IP addresses should not be the same. If without SNAT, response packet will not be
+forwarded back to OVS pipeline as the destination IP is the Pod's own IP, then the
+connection is interrupted because the conntrack state is only stored in OVS ct zone,
+not in the Pod. With SNAT, the destination IP will be the virtual hairpin IP and
+forwarded back to OVS pipeline. Note that, bit 18 in NXM_NX_REG0 is set to 0x1, and
+it is consumed in table [L2ForwardingOutTable] to output the packet on the port on
+which it was received with action `IN_PORT`. Action `IN_PORT` is the only standard
```

Comment:
Remove "Action `IN_PORT` is the only...".

Reply:
Thanks, removed

Reply:
Updated.